pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -275,3 +273,15 @@ def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
 
         Use ``load_state_dict`` instead.
         """
+
+    @property
+    def precision(self) -> Union[str, int]:
+        return 32
+
+    @property
+    def amp_backend(self) -> Optional[AMPType]:
+        return None
+
+    @property
+    def amp_level(self) -> Optional[str]:
+        return None
```

Comment:
I'm not sure it make sense to include this in the base class. 
The TPU, IPU, DoublePrecision (which inheritance from the base class PrecisionPlugin) doesn't have amp. I think we can add in MixPrecisionPlugin. Maybe deepSpeedPrecisionPlugin can inherent from from MixPrecisionPlugin as well?

Overall idea is similar to ParallelStrategy vs Strategy. If it's parallel specific property, we only included them in ParallelStrategy instead of the base class

Reply:
That makes sense! Should I first send out a PR to make DeepSpeedPrecisionPlugin inherit from MixedPrecisionPlugin?

Reply:
Actually one issue with this is we have the `amp_backend` property on the Trainer which references `precision_plugin.amp_backend`. I guess we could have it just first check if the precision plugin is of type MixedPrecisionPlugin and if not return None. what do you think is the best way to go about this?